src/main/java/dev/waterdog/scheduler/WaterdogScheduler.java

```diff
@@ -26,7 +26,7 @@
 
 public class WaterdogScheduler {
 
-    private static WaterdogScheduler instance;
+    private static final WaterdogScheduler INSTANCE = new WaterdogScheduler(ProxyServer.getInstance());
```

Comment:
At this point `ProxyServer.getInstance()` will return null. Also don't forget about lowercase😉

Reply:
The proxy instance gets initiated before the waterdog scheduler so it cant be null

Reply:
Scheduler instance is initialized statically so it is initialized before proxy starts.

Reply:
No the static method gets called when the class loader loads it and this happens when a static method gets executed, an instance gets created or the class gets loaded with `Class.forName()`

Reply:
I could create a test so the build would fail if the instance is null

Reply:
Alright. I see what you mean now. Also I'm thinking about allowing plugins to register create own scheduler which would extend `WaterdogScheduler` class so instance would be created after plugins are loaded.